src/sourmash/lca/command_summarize.py

```diff
@@ -117,31 +114,21 @@ def output_results(lineage_counts, total_counts, filename=None, sig=None):
         p = count / total_counts * 100.
         p = '{:.1f}%'.format(p)
 
-        if filename and sig:
-            print_results('{:5} {:>5}   {}   {}:{} {}'.format(p, count, lineage, filename, sig.md5sum()[:8], sig))
-        else:
-            print_results('{:5} {:>5}   {}'.format(p, count, lineage))
-
+        print_results('{:5} {:>5}   {}   {}:{} {}'.format(p, count, lineage, filename, sig.md5sum()[:8], sig))
 
 def output_csv(lineage_counts, csv_fp, filename, sig, write_header=True):
     """\
     Output results in CSV.
     """
-    if filename or sig:                   # require both
-        assert filename and sig
 
     w = csv.writer(csv_fp)
     if write_header:
         headers = ['count'] + list(lca_utils.taxlist())
-        if filename:
```

Comment:
huh.... doesn't this remove that information (filename, sig_name, sig_md5) from the output? Surprised that doesn't break the tests!?

Reply:
ok, checked on this, and yes this does change the CSV output, and no it doesn't break any tests. <sigh>

so, I think we need to add tests for the CSV output to make sure it's got all the right info (filename, sig_name, and sig_md5). Do you want to take this on, or would you like me to add them?

Reply:
I'd like to learn how to write tests and could use pointers for how to start!

Reply:
ok! here's some starting suggestions -

1. in `tests/test_lca.py`, copy the entire function `def test_single_summarize_to_output():` and rename the copy to `test_single_summarize_to_output_check_filename` or something
2. run the equivalent sourmash command at the command line (something like `sourmash lca summarize --db tests/test-data/lca/delmont-1.lca.json --query tests/test-data/lca/TARA_ASE_MAG_00031.sig -o output.txt`) and then examine `output.txt` to see what headers and content it has
3. change the test code for `cmd` to use `'q.sig'` instead of `input_sig`, so that you have a specific filename to look that is not dependent on your development setup;
4. adjust the second assert statement to match expected output from above
5. add an assert statement that confirms that the expected columns are in the header of the `output.txt` file also

you might do this on a separate branch, forked from `latest`; then, once you get the tests working, you can merge into your PR and they should fail again (until you adjust your CSV output code). This is one way I make sure that the tests actually test something :)

Reply:
I think I got parts of the above suggestions working (I did get the headers and content from output.txt), but not sure about the 3rd suggestion. I changed `input_sig` to `'q.sig'` in the `cmd` line but get error that the path is incorrect. 
```
cmd = ['lca', 'summarize', '--db', db1, '--query', 'q.sig',
               '-o', os.path.join(location, 'output.txt')]
```

error: 
```
KError! Could not find the following files. Make sure the file paths are specified correctly.
E             q.sig
```

Reply:
ahh! my mistake. add `in_location=location` as a parameter to `runscript(...)`.

You can also look at what's going on with `runcmd` and factoring out `utils.TempDirectory(...)` but that's even further afield :)

Reply:
the tests are working locally! and i think the tests are failing on #1544 as expected because i need to add back 2 lines on this PR.

Reply:
👍